src/main/java/zd/zero/waifu/motivator/plugin/motivation/MotivationFactory.kt

```diff
@@ -88,6 +94,9 @@ object MotivationFactory {
         motivationConstructor: (MotivationAsset) -> WaifuMotivation,
         assetSupplier: () -> Optional<MotivationAsset>
     ) {
+        motivationStatistics.registerEvent(motivationEvent)
+        if (shouldShowAlert(motivationEvent).not()) return
+
         ApplicationManager.getApplication().executeOnPooledThread {
```

Comment:
> However, another part of me is like, well they are causing these events, so they should directly be affecting the mood state of the plugin. Which gives use the opportunity to do more interesting things in the future.

I agree with this, as these events happen. So I'm thinking of changing the user story description to cater to this scenario. I'm proposing to 

If the moods are the same, that's when we start counting, if the mood is always `HAPPY` we'll only show it for the set number of invocations, but whenever the mood changes then we'll ignore the count and immediately show the event, so a changing mood will always show the alerts, but repeating moods will utilize the configuration, this, of course, will change the descriptions to make it relevant to this one.

My main motivation is whenever a user works in a TDD environment, there would be a lot of multiple failures sequentially, so when a test passes there should now an alert is shown and vice versa. wdyt?

Reply:
> whenever the category changes...My main motivation is whenever a user works in a TDD environment

I like the direction that you are taking this however, I think there might be a better solution to this, in my opinion.

Rather than always counting the invocations, how about we have like a `Minimal Mode`. Where, for a brief period of time, we only receive an alert when the `MotivationEventCategory` changes (eg. test passing/failing, or fixing your broken build). After that period expires, then the plugin goes back to alerting to any event.

I've got more ideas about this, but I wanted to get your opinion on this, because it is a bit different than what was initially proposed. I actually have use-cases for this, and would like to have this implemented.

Reply:
Oh I see, this might be related to #231 but in that story, there would be no alerts at all for a period of time. We should however throw away this issue and the more I think of the use case for #106 there are more better way we can achieve for the intended motivation.

Then we'll have a plugin where we have both:
* #231 For a timed no alerts at all and a
* `Minimal Mode` which only alerts whenever the category/mood changes, wdyt?

Reply:
> there are more better way we can achieve for the intended motivation.

I think we both understand the current use case, and I think that the intention for this issue was good. However, from using this plugin for a while know, there has only been times where I would briefly disable reactions, because I knew I was going to trigger a bunch of events. After, that, I would turn it back on, and enjoy the steady stream of reactions again.

I am in favor of re-imagining the current issue to give a better user experience. Sorry for bringing this up, after you did the work :sweat_smile:

Reply:
No that's fine, it is better to drop things earlier if they are not relevant anymore, I don't get attached to the codes I write 😂. Better *Waifu* experience is the priority. I'll go ahead and close the PRs and related issues, and create a new one for the discussed story.